theano/tensor/opt.py

```diff
@@ -6919,6 +6920,43 @@ def local_add_mul_fusion(node):
                            'FusionOptimizer')
 
 
+@register_canonicalize
+@gof.local_optimizer([Elemwise])
+def local_useless_composite(node):
+    """For elemwise Composite that have multiple outputs, remove the
+    outputs that are not used and if there is only 1 node in the
+    Composite, remove it in some cases.
+
+    """
+    if (not isinstance(node.op, Elemwise) or
+            not isinstance(node.op.scalar_op, scalar.Composite)):
+        return
+    comp = node.op.scalar_op
+
+    # Remove outputs not used.
+    idx = [i for i, o_extern in enumerate(node.outputs)
+           if o_extern.clients]
+    if len(idx) < len(node.outputs):
+        new_outputs = [comp.outputs[i] for i in idx]
+        c = scalar.Composite(inputs=comp.inputs,
+                             outputs=new_outputs)
+        e = Elemwise(scalar_op=c)(*node.inputs, return_list=True)
+        comp = e[0].owner.op.scalar_op
```

Comment:
Shouldn't we try to get rid of useless inputs too?  Unless it already does that.